game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -286,7 +284,7 @@ private void sortTransportsToUnload(final List<Unit> transports, final Route rou
     sortUnitsToMove(allUnitsInSelectedTransports, route);
     final List<Unit> selectedUnitsToUnload = new ArrayList<>();
     final List<Unit> sortedTransports = new ArrayList<>(chosenTransports);
-    sortedTransports.sort(UnitComparator.getIncreasingCapacityComparator(sortedTransports));
+    sortedTransports.sort(UnitComparator.getIncreasingCapacityComparator());
```

Comment:
Good question didn't notice the comparator could potentially be re-used.
I'll have a look at the values that are being cached and will then decide if it's "dangerous" to reuse the comparator

Reply:
I looked at the code and if my assumption is correct that `MovePanel#getUnitsToUnload` doesn't actually alter the `GameData`, it should be safe to reuse the same Comparator for both operations.
The reason why it was done this way is probably because there are actually 2 variables called `sortedTransports`, one is inside a lambda, so the one inside the lambda couldn't know all the final values in advance, so it had to calculate the comparator once it knew the values.
I'm not 100% sure on this one though, that's why I will test a reusal of his comparator in a seperate PR.
I'll also have a look at `TripleAUnit#getTransporting` which currently has a runtime of `O(n)` because it needs to iterate through all territories on the map to find the territory the current instance is part of.
A simple backreference to the territory might already do the trick.
This could potentially reduce the total runtime of the total compare value calculations from `O(n²)` to `O(n)`
Will have a look at this